sdk/tdf_config.go

```diff
@@ -43,6 +44,101 @@ type KASInfo struct {
 	Default bool
 }
 
+type AssertionSigningKeyType uint
+
+const (
+	AssertionSigningKeyRS256PubKey AssertionSigningKeyType = iota + 1
+	AssertionSigningKeyRS256PriKey
+	AssertionSigningKeyHS256PayloadKey
+	AssertionSigningKeyHS256UserDefined
+)
+
+var (
+	assertionSigningKeyTypeName = map[uint8]string{
+		uint8(AssertionSigningKeyRS256PubKey):      "RS256PublicKey",
+		uint8(AssertionSigningKeyRS256PriKey):      "RS256PrivateKey",
+		uint8(AssertionSigningKeyHS256PayloadKey):  "HS256PayloadKey",
+		uint8(AssertionSigningKeyHS256UserDefined): "HS256UserDefined",
+	}
+	assertionSigningKeyTypeValue = map[string]uint8{
+		"RS256PublicKey":   uint8(AssertionSigningKeyRS256PubKey),
+		"RS256PrivateKey":  uint8(AssertionSigningKeyRS256PriKey),
+		"HS256PayloadKey":  uint8(AssertionSigningKeyHS256PayloadKey),
+		"HS256UserDefined": uint8(AssertionSigningKeyHS256UserDefined),
+	}
+)
+
+func (a AssertionSigningKeyType) String() string {
+	return assertionSigningKeyTypeName[uint8(a)]
+}
+
+func ParseAssertionSigningKeyType(a string) (AssertionSigningKeyType, error) {
+	a = strings.TrimSpace(a)
+	value, ok := assertionSigningKeyTypeValue[a]
+	if !ok {
+		return AssertionSigningKeyType(0), fmt.Errorf("%q is not a valid assertion signing key type", a)
+	}
+	return AssertionSigningKeyType(value), nil
+}
+
+type TDFAssertionOption func(*AssertionConfig) error
+
+type AssertionConfig struct {
+	keyType AssertionSigningKeyType
+	payload []byte // If the type
+}
```

Comment:
I was thinking the assertion config would be a subset of fields we want the user to set. Because right now we have the WithAssertion option which takes an assertion allowing a user to set the binding themselves. Which I'm not sure we want.

@biscoe916 Any thoughts on this?

Reply:
The user should not set the binding. I'll make it private

Reply:
That is confusing from a developer experience perspective. That's why I'm suggesting changes.